### ISSUE TITLE
Cron job is running twice

### ISSUE THREAD

**Issue by ichbinedgar** [liebig/cron#67]
Hi!
I hope you are doing well. I really tried to find answers here in your issues, but I could not find any solution yet. So thats why I opened this issue.
 
Problem description:
I have a cron job that sends a test email to my personal inbox, and the problem is that I've received the email twice. I'm using Laravel 4.2.

So, in order to see what was the problem, I defined my crontab like this:
* * * * * /opt/lampp/bin/php /opt/lampp/htdocs/myapp/artisan cron:run <- it runs every minute

and my global.php is like this:
```php

Event::listen('cron.collectJobs', function() {
	
	\Cron::add('test', '15 17 * * *', function() {
        // sends email at 17:15 UTC time	
		Mail::send('emails/auth/reminder', array(), function($message)
		{
			$message->from('app@app.com', 'gfdgf');

			$message->to('myemail@orgexample.com')->subject('Welcome to My Laravel app! morning!!');
		});
		return null;
    });

   /*\Cron::remove('morning_mail');*/
	
	$report = \Cron::run();	
	
});

```
   
In theory, or what I expect is that I will receive just one email at 17:15 UTC time, but I received two. 

In the database, the cron_manager table is saving duplicated entries , for example:

IDx1 | 2015-06-04 17:54:01 |  0.00
idx2  | 2015-06-04 17:54:01 |  0.00
 
and it's happening the same issue on the cron_job table as well.

I've read this comment (https://github.com/liebig/cron/issues/52) :

"Have a look at the cron_manager and cron_job database tables. If you find more than one entry in the cron_manager table for your defined time interval, you probably execute Cron too often. If you find more than one entry in the cron_job table for a manager id, your probably call your send-mail-function too often. Otherwise have a look at your send-mail-function. Maybe it sends the mail twice?"

It's something very similar to my case, but I don't know how to avoid execute cron too often. I tried to only excute the cronjob at some specific time and it didn't work. Also, as you saw in my previous code, I think the issue is not related to the send-mail-function.

Well, thats it. Let me know if my explanation was not clear. 
Thank you!!! :)

**Comment by liebig** [comment]
Hi,
Thank you for using Cron. I am sorry about your problem with Cron and I will try to help you.

First of all you can test the prevent overlapping function. Just add `\Cron::setEnablePreventOverlapping();` to the `global.php` file above the `Event::listen` command. But we have to find the reason for the twice call. Let's try to change your expression at your `\Cron::add` to every minute (`* * * * *`) and use `var_dump` to print the `$report` variable. After that try to call the route from console (`/opt/lampp/bin/php /opt/lampp/htdocs/myapp/artisan cron:run`) and please copy the return to this issue.
Your code should look like this:
```php
Event::listen('cron.collectJobs', function() {

    \Cron::add('test', '* * * * *', function() {
        // sends email at 17:15 UTC time    
        Mail::send('emails/auth/reminder', array(), function($message)
        {
            $message->from('app@app.com', 'gfdgf');

            $message->to('myemail@orgexample.com')->subject('Welcome to My Laravel app! morning!!');
        });
        return null;
    });

   /*\Cron::remove('morning_mail');*/

    $report = \Cron::run(); 
	var_dump($report)
});
```

Thank You
Marc

**Comment by liebig** [comment]
I just notice, that your crontab expression looks wrong: `* * * * /opt/lampp/bin/php /opt/lampp/htdocs/myapp/artisan cron:run`. The expression should have 5 asterisks. It should look luke: `* * * * /opt/lampp/bin/php /opt/lampp/htdocs/myapp/artisan cron:run`. 
Please test this.

Thank you
Marc

**Comment by ichbinedgar** [comment]
Hi liebig! Thanks for you support.

I tried to use "\Cron::setEnablePreventOverlapping();" but it didn't work. Actually, I received three emails intstead of two.
The crontab expression is OK. I did not notice that the editor replaced the first star/asterisk with a bullet.

This is the ouput of the var_dump:

```
array(6) {
  ["rundate"]=>
  int(1433513448)
  ["inTime"]=>
  bool(true)
  ["runtime"]=>
  float(0.047616004943848)
  ["errors"]=>
  int(0)
  ["crons"]=>
  array(1) {
    [0]=>
    array(3) {
      ["name"]=>
      string(4) "test"
      ["return"]=>
      NULL
      ["runtime"]=>
      float(0.047450065612793)
    }
  }
  ["lastRun"]=>
  array(2) {
    ["rundate"]=>
    string(19) "2015-06-05 14:10:01"
    ["runtime"]=>
    float(0.0099999997764826)
  }
}
+------------+---------+----------+--------+------+
| Run date   | In time | Run time | Errors | Jobs |
+------------+---------+----------+--------+------+
| 1433513448 | false   | 0.0153   | 0      | 1    |
+------------+---------+----------+--------+------+
```
Can you see something weird?
Thanks

**Comment by liebig** [comment]
Hmm when you execute the job from command shell, does it run twice, too? Can you post the entries of the database tables `cron_job` and `cron_manager` and please post your log file. There should be more information. The only thing that is weird with your output is the `In time` parameter which is in the `var_dump` output true and in the console output false.
Sorry for the inconvenience!

**Comment by ichbinedgar** [comment]
Hi liebig
When I excute the job, it did not seems to be running twice. 
I decided to postpone the use of your library for the moment, because I don't need something complex at the moment and it's kind of tricky to solve this issue. 
I decided to use the following simple cronjob : 
 * * * * * *  wget -O - http://url.com/script_page >/dev/null 2>&1
And thats what I need for the moment, but thank you for you support. I hope I can use it next time. You can close this issue.

**Comment by liebig** [comment]
Hi,
Maybe there is a problem with your crontab. If the execution of the job via console works I don't know why the job via crontab runs twice. Cron has a build in route, too. So after configuration you can use wget with the route `http://yourdomain.com/cron.php?key=securitykey`.

**Comment by liebig** [comment]
Just for completeness: The problem is this `$report = \Cron::run();`. If you use the artisan command or the integrated route you don't have to execute `\Cron::run()`. The command and route do this after firing the `cron.collectJobs` event for you. So everything works fine. By the way, the overlapping prevention doesn't prevent this, because the `\Cron::run()` method is called successive, not in parallel.

I am very sorry that I overlooked this.

**Comment by ichbinedgar** [comment]
Hi liebig! 
Nice, you figured it out, don't worry.
Thanks for let me know  :)
